content/tutorials/max-gas-pub-data/10.index.md

```diff
@@ -0,0 +1,421 @@
+        await sendAndExplainTx("100", "2000000"); // Very low gasPerPubdata, 
+        await sendAndExplainTx(utils.DEFAULT_GAS_PER_PUBDATA_LIMIT, "100000000"); // Very high gasLimit
+    
+        // // Get and log the transaction receipt for the vote
+        // const receipt = await deployContract
+      } catch (error) {
+        console.error("Deployment or interaction failed: ", error);
+        process.exitCode = 1;
+      }
+    }
+    ```
+::
+
+This deployment script serves as a practical example of how to interact with ZKsync Era, showcasing the impact
+of different gas settings on transaction execution and costs.
+
+This script deploys the contract and tests scenarios with varying `gasPerPubdata` and `gasLimit` values.
+
+**The script includes:**
+
+1. Deploy the `ZKFestVoting` contract.
+2. Execute a series of test votes with different `gasPerPubdata` and `gasLimit` settings.
+3. Log the results of each transaction, including gas usage and any errors encountered.
+
+## How It Works
+
+1. **Transaction Submission**: When sending a transaction, specify the max gas per pubdata.
+
+2. **Execution**: ZKsync Era executes the transaction and calculates the actual pubdata cost.
+
+3. **Comparison**: The actual cost is compared against your specified max value.
+
+4. **Outcome**:
+
+- If actual cost ≤ specified max: Transaction succeeds
+- If actual cost > specified max: Transaction is rejected
+  - Which in this case, it's happening with the very low gasPerPubdata example `await sendAndExplainTx("100", "2000000");`
+
+## Key Aspects of Gas Measurement and DEFAULT_GAS_PER_PUBDATA_LIMIT
+
+- **Deployment Fee Estimation**:
+The script estimates the deployment fee before deploying the contract, giving insight into the initial cost.
+- **Custom Transaction Creation**:
+The `createCustomTx` function allows custom `gasPerPubdata` and `gasLimit` values to be set for each transaction.
+- **Testing Different Scenarios**:
+  - failsDefault `DEFAULT_GAS_PER_PUBDATA_LIMIT` with a moderate `gasLimit`
+  - Very low `gasPerPubdata` to see how it affects transaction execution
+  - Default `DEFAULT_GAS_PER_PUBDATA_LIMIT` with a very high `gasLimit`
+- **Transaction Monitoring**:
+The script logs transaction details, including gas used and any errors encountered.
+- **Error Handling**:
+Comprehensive error handling and logging provide insights into transaction failures.
+
+## Key Takeaways
+
+1. The `DEFAULT_GAS_PER_PUBDATA_LIMIT` (accessed via zksync-ethers library on `utils.DEFAULT_GAS_PER_PUBDATA_LIMIT`)
+serves for setting default pubdata gas limits.
+2. Very low gas per pubdata values may lead to transaction rejection due to insufficient pubdata gas.
+    - In this case, the transaction won’t be included in the blockchain, and the user is not charged anything.
+3. Extremely high `gasLimit` values may not necessarily improve transaction success rates but could lead to higher costs.
```

Comment:
better to say "upfront costs", i.e. the final cost is the same, but the upfront costs are increased by higher gaslimit yes